cpp/src/join/conditional_join.cu

```diff
@@ -96,10 +97,6 @@ std::unique_ptr<rmm::device_uvector<size_type>> conditional_join_anti_semi(
     join_size = size.value(stream);
   }
 
-  if (left.num_rows() == 0) {
-    return std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr);
-  }
```

Comment:
This code path is unreachable. The case of zero rows is handled earlier in the function.